diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -0,0 +1,195 @@
+        checkOpenBraceOnSameLine(node, LBRACE, configuration)
+        checkCloseBrace(node, configuration)
+    }
+
+    private fun checkFun(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
+        if (node.hasChildOfType(BLOCK)) {
+            checkOpenBraceOnSameLine(node, BLOCK, configuration)
+            checkCloseBrace(node.findChildByType(BLOCK)!!, configuration)
+        }
+    }
+
+    private fun checkIf(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
+        val ifPsi = node.psi as KtIfExpression
+        val thenNode = ifPsi.then?.node
+        val hasElseBranch = ifPsi.elseKeyword != null
+        val elseNode = ifPsi.`else`?.node
+        if (thenNode != null && thenNode.hasChildOfType(LBRACE)) {
+            checkOpenBraceOnSameLine(node, THEN, configuration)
+            checkCloseBrace(thenNode, configuration)
+        }
+        if (hasElseBranch && elseNode!!.elementType != IF && elseNode.hasChildOfType(LBRACE)) {
+            checkOpenBraceOnSameLine(node, ELSE, configuration)
+            checkMidBrace(node)
+            checkCloseBrace(elseNode, configuration)
+        }
+    }
+
+    private fun checkOpenBraceOnSameLine(node: ASTNode, beforeType: IElementType, configuration: BlockStructureBracesConfiguration) {
+        if (!configuration.openBrace) return
+        val braceSpace = node.findChildByType(beforeType)?.treePrev
+        if (braceSpace == null || braceSpace.elementType != WHITE_SPACE || braceSpace.text.contains("\n".toRegex()))
+            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect newline before opening brace",
+                    (braceSpace ?: node).startOffset) {
+            }
+        checkOpenBraceEndLine(node, beforeType)
+    }
+
+    private fun checkOpenBraceEndLine(node: ASTNode, beforeType: IElementType){
+        val newNode = when (node.elementType) {
+            IF -> {
+                when (beforeType) {
+                    THEN -> node.findChildByType(THEN)?.findChildByType(BLOCK)?.findChildByType(LBRACE)?.treeNext
+                    else -> node.findChildByType(ELSE)?.findChildByType(BLOCK)?.findChildByType(LBRACE)?.treeNext
+                }
+            }
+            WHEN -> node.findChildByType(LBRACE)?.treeNext
+            FOR, WHILE, DO_WHILE -> node.findChildByType(BODY)?.findChildByType(BLOCK)?.findChildByType(LBRACE)?.treeNext
+            CLASS, OBJECT_DECLARATION -> node.findChildByType(CLASS_BODY)!!.findChildByType(LBRACE)?.treeNext
+            else -> node.findChildByType(BLOCK)?.findChildByType(LBRACE)?.treeNext
+        }
+        if (newNode == null || newNode.elementType != WHITE_SPACE || !newNode.text.contains("\n")) {
+            BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect same line after opening brace",
+                    (newNode ?: node).startOffset) {
+            }
+        }
+    }
+
+    private fun checkMidBrace(node: ASTNode) {
```

Comment:
Do you really need to look for keywords here? You already handle `else`, `catch` and `finally` in other checks, so you could, for example, pass `elseNode.treePrev` from `checkIf` and only check for correct newline inside `checkMidBrace` method.